searchcore/src/vespa/searchcore/proton/server/lid_space_compaction_job_take2.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "lid_space_compaction_job_take2.h"
+#include "i_document_scan_iterator.h"
+#include "i_lid_space_compaction_handler.h"
+#include "i_operation_storer.h"
+#include <vespa/searchcore/proton/feedoperation/moveoperation.h>
+#include <vespa/persistence/spi/bucket_tasks.h>
+#include <vespa/vespalib/util/destructor_callbacks.h>
+#include <cassert>
+
+using search::DocumentMetaData;
+using search::LidUsageStats;
+using storage::spi::makeBucketTask;
+using storage::spi::Bucket;
+
+namespace proton::lidspace {
+
+bool
+CompactionJob::scanDocuments(const LidUsageStats &stats)
+{
+    if (_scanItr->valid()) {
+        DocumentMetaData document = getNextDocument(stats, false);
+        if (document.valid()) {
+            Bucket metaBucket(document::Bucket(document::BucketSpace::placeHolder(), document.bucketId));
```

Comment:
We might need a proper bucket space here.

Reply:
Yes, that is correct. Where is that knowledge ?

Reply:
`MaintenanceJobsInjector::injectJobs` gets it as an argument.

Reply:
K, found it, fixed.